test/test_dataloader.py

```diff
@@ -62,6 +63,13 @@
     HAS_DILL = False
 skipIfNoDill = unittest.skipIf(not HAS_DILL, "no dill")
 
+
+try:
+    import numpy as np
+except ImportError:
+    np = None
```

Comment:
this is not very helpful, as tests will break anyway, better add skipping mechanism

Reply:
I later check if `np` is available before attempting to use it, and use apply a non-`np`, equivalent transformation instead.

https://github.com/pytorch/pytorch/pull/72896/files#diff-7cd4c7412591dcd9a7f8b7eb6622e286d9c19e3835bc8319af2ab4a1cf2e7badR1380

Other options:
1. Import it within the test (I was worried that will make `np` non-global and accidentally pass a test that should fail
2. Skip when `np` is not available

Reply:
But it will correctly work only if np exists, otherwise, even old code will pass this test.
Also because `np` here is global, you are putting landmine under all users of `np` here, as they would have (and sometimes forget) to do same `if np in not None` check.

Reply:
I see, that makes sense. I will skip the test if `np` is missing then.

Reply:
This is done